pennylane/ops/op_math/decompositions/unitary_decompositions.py

```diff
@@ -335,114 +506,99 @@ def _decomposition_1_cnot(U, wires):
+    ops.QubitUnitary(C, wires=wires[0])
+    ops.QubitUnitary(D, wires=wires[1])
+    ops.CNOT(wires=wires)
+    ops.QubitUnitary(A, wires=wires[1])
+    ops.QubitUnitary(B, wires=wires[0])
 
-    return C_ops + D_ops + [qml.CNOT(wires=wires)] + A_ops + B_ops + [qml.GlobalPhase(np.pi / 4)]
+    return -np.pi / 4
 
 
-def _decomposition_2_cnots(U, wires):
+def _decompose_2_cnots(U, wires):
     r"""If 2 CNOTs are required, we can write the circuit as
      -╭U- = -A--╭X--RZ(d)--╭X--C-
      -╰U- = -B--╰C--RX(p)--╰C--D-
     We need to find the angles for the Z and X rotations such that the inner
     part has the same spectrum as U, and then we can recover A, B, C, D.
     """
-    # Compute the rotation angles
 
-    u = math.dot(Edag, math.dot(U, E))
+    # Compute the rotation angles
+    u = math.dot(E_dag, math.dot(U, E))
     gammaU = math.dot(u, math.T(u))
-    evs, _ = math.linalg.eig(gammaU)
+    evs = math.linalg.eigvals(gammaU)
 
     # These choices are based on Proposition III.3 of
     # https://arxiv.org/abs/quant-ph/0308045
-    # There is, however, a special case where the circuit has the form
-    # -╭U- = -A--╭C--╭X--C-
-    # -╰U- = -B--╰X--╰C--D-
-    #
-    # or some variant of this, where the two CNOTs are adjacent.
-    #
-    # What happens here is that the set of evs is -1, -1, 1, 1 and we can write
-    # -╭U- = -A--╭X--SZ--╭X--C-
-    # -╰U- = -B--╰C--SX--╰C--D-
-    # where SZ and SX are square roots of Z and X respectively. (This
-    # decomposition comes from using Hadamards to flip the direction of the
-    # first CNOT, and then decomposing them and merging single-qubit gates.) For
-    # some reason this case is not handled properly with the full algorithm, so
-    # we treat it separately.
-
-    sorted_evs = math.sort(math.real(evs))
-
-    if math.allclose(sorted_evs, [-1, -1, 1, 1]):
-        interior_decomp = [
-            qml.CNOT(wires=[wires[1], wires[0]]),
-            qml.S(wires=wires[0]),
-            qml.SX(wires=wires[1]),
-            qml.CNOT(wires=[wires[1], wires[0]]),
-        ]
 
-        # S \otimes SX
-        inner_matrix = S_SX
-    else:
-        # For the non-special case, the eigenvalues come in conjugate pairs.
-        # We need to find two non-conjugate eigenvalues to extract the angles.
-
-        x = math.angle(evs[0])
-        y = math.angle(evs[1])
-
-        # If it was the conjugate, grab a different eigenvalue.
-        if math.allclose(x, -y):
-            y = math.angle(evs[2])
-
-        delta = (x + y) / 2
-        phi = (x - y) / 2
-        interior_decomp = [
-            qml.CNOT(wires=[wires[1], wires[0]]),
-            qml.RZ(delta, wires=wires[0]),
-            qml.RX(phi, wires=wires[1]),
-            qml.CNOT(wires=[wires[1], wires[0]]),
-        ]
+    x = math.angle(evs[0])
+    y = math.angle(evs[1])
+
+    # If it was the conjugate, grab a different eigenvalue.
+    y = math.cond(math.allclose(x, -y), lambda: math.angle(evs[2]), lambda: y, ())
```

Comment:
```suggestion
    y = math.cond(math.allclose(evs[0], math.conj(evs[1])), lambda: math.angle(evs[2]), lambda: y, ())
```
The original condition for testing conjugates is simply wrong by not considering the conjugate between -1 and -1. This is the simplest way of a correct conjugate testing.

Reply:
Note that this won't fix `test_two_qubit_decomposition_special_case_discontinuity` (one can check by setting the integer multiples for `RZd` and `RXp` below to zero to repeat the failure). Must be lots of bug with this script! :anger:

Reply:
Yeah it seems that with or without the `eps`, the eigenvectors returned by 
```python
q = math.dot(q, math.diag([1, 1, 1, math.sign(math.linalg.det(q))]))
```
are in different orders.